### ISSUE TITLE
UK/EU Time Format Option + dd MMM yy Format Option

### ISSUE THREAD

**Issue by Jazztache** [RandomGamingDev/yt-times#2]
Hello!

Can you please add the option for DD/MM/YY time format, as well as dd MMM yy (for example 16 Nov24)?

Thankyou!

**Comment by RandomGamingDev** [comment]
Without adding a library adding dynamic formatting for basic things like choosing between the different ways you can display the months would be pretty annoying, and it wouldn't consider locales, so I decided to create a basic date formatter alongside customizable options for `toLocaleString()`.

![image](https://github.com/user-attachments/assets/6ededf2d-1d9f-40c0-a51b-5d579ad499e9)

Everything except `format`'s options for `toLocaleString()`.

Would u say this fulfills what you wanted?

**Comment by blooddiamondkim** [comment]
Hey there, jumping in from your comment on the review I left on Firefox extensions. Are we able to pick the order? I think the order would be useful, since my request and the request of others were specifically changing up the order in addition to the formatting

**Comment by RandomGamingDev** [comment]
> Hey there, jumping in from your comment on the review I left on Firefox extensions. Are we able to pick the order? I think the order would be useful, since my request and the request of others were specifically changing up the order in addition to the formatting

Yes, you would, but only in `formatDate` mode unless it's the format you want's already some locale's format.

**Comment by blooddiamondkim** [comment]
> > Hey there, jumping in from your comment on the review I left on Firefox extensions. Are we able to pick the order? I think the order would be useful, since my request and the request of others were specifically changing up the order in addition to the formatting
> 
> Yes, you would, but only in `formatDate` mode unless it's the format you want's already some locale's format.

Cool. In that case I think it would absolutely fulfill what I was personally hoping for. I appreciate that you're working on expanding the capabilities of the add-on.

**Comment by RandomGamingDev** [comment]
> > > Hey there, jumping in from your comment on the review I left on Firefox extensions. Are we able to pick the order? I think the order would be useful, since my request and the request of others were specifically changing up the order in addition to the formatting
> > 
> > 
> > Yes, you would, but only in `formatDate` mode unless it's the format you want's already some locale's format.
> 
> Cool. In that case I think it would absolutely fulfill what I was personally hoping for. I appreciate that you're working on expanding the capabilities of the add-on.

alr great. I'll be busy for quite a bit and would like to see if I could get more comments from other people who requested the same feature to see if it'll fulfill all of their requests or if it's say, too complex. This should hopefully be done within a week or two.

**Comment by AceFrahm** [comment]
This extension says it will use the system locale, but it doesn't do that on my machine.  It should give YYYY-DD-MM, but instead it's presenting time as "12/12/2022, 2:49:18 PM", the WORST of all possible formats, because it's ambiguous as to whether or not the day or month is first, and it fails to use the 24 hour time format, which forces the brain to do extra unnecessary work processing "day or night" when you'd rather be simply comparing numbers to determine which video was posted first.  

The only way to ensure items can be time-sorted in order when the text is cut & pasted elsewhere is use yyyy-MM-dd HH:mm:ss.  This is the ONLY universally good format, because it's the only that is not ambiguous and does not omit data fields.  It's the only one that puts the digits in order from MOST significant to *least* significant, and provides zero-padding for the month & day numbers so the text will align when the data is in a column together.

I don't know why it doesn't follow my local date-time settings on windows 10 in firefox.  I don't see any settings in this extension or inside firefox that can change the behavior.

**Comment by RandomGamingDev** [comment]
> This extension says it will use the system locale, but it doesn't do that on my machine. It should give YYYY-DD-MM, but instead it's presenting time as "12/12/2022, 2:49:18 PM", the WORST of all possible formats, because it's ambiguous as to whether or not the day or month is first, and it fails to use the 24 hour time format, which forces the brain to do extra unnecessary work processing "day or night" when you'd rather be simply comparing numbers to determine which video was posted first.
> 
> The only way to ensure items can be time-sorted in order when the text is cut & pasted elsewhere is use yyyy-MM-dd HH:mm:ss. This is the ONLY universally good format, because it's the only that is not ambiguous and does not omit data fields. It's the only one that puts the digits in order from MOST significant to _least_ significant, and provides zero-padding for the month & day numbers so the text will align when the data is in a column together.
> 
> I don't know why it doesn't follow my local date-time settings on windows 10 in firefox. I don't see any settings in this extension or inside firefox that can change the behavior.

Would you be able to send me what locale is used on your machine (you should be able to find it in settings)?

**Comment by RandomGamingDev** [comment]
> Hello!
> 
> Can you please add the option for DD/MM/YY time format, as well as dd MMM yy (for example 16 Nov24)?
> 
> Thankyou!

@Jazztache DD/MM/YY has been added I'll see if I have the time to add dd MMM yy as well, although I don't see too much use for them (especially since other languages won't use the same names and numbers represent them well enough.)

**Comment by shapirus** [comment]
> > Hello!
> > Can you please add the option for DD/MM/YY time format, as well as dd MMM yy (for example 16 Nov24)?
> > Thankyou!
> 
> [@Jazztache](https://github.com/Jazztache) DD/MM/YY has been added I'll see if I have the time to add dd MMM yy as well, although I don't see too much use for them (especially since other languages won't use the same names and numbers represent them well enough.)

Is there a possibility to let the user specify an `strftime()`-compatible format? That would solve the issue completely for everyone.


For more input, I'm also having an issue with the extension not following the system locale.

For example, in my case it is using a US date format: `7/14/2025, 12:39:37 PM`, even though the LC_TIME env variable is set to "C":

```
$ locale
LANG=en_US.UTF-8
LANGUAGE=en_US:en
LC_CTYPE="en_US.UTF-8"
LC_NUMERIC="en_US.UTF-8"
LC_TIME=C
LC_COLLATE="en_US.UTF-8"
LC_MONETARY="en_US.UTF-8"
LC_MESSAGES="en_US.UTF-8"
LC_PAPER="en_US.UTF-8"
LC_NAME="en_US.UTF-8"
LC_ADDRESS="en_US.UTF-8"
LC_TELEPHONE="en_US.UTF-8"
LC_MEASUREMENT=C
LC_IDENTIFICATION="en_US.UTF-8"
LC_ALL=
```

The `date` utility does work correctly (seen by the 24-hour format), e.g.:

```
$ date
Mon Jul 14 13:12:15 EEST 2025
```

And to verify it, it does switch to the 12-hour format if the time locale is set to US explicitly:

```
$ LC_TIME=en_US.UTF-8 date
Mon Jul 14 01:15:38 PM EEST 2025
```

It's always better to give the user an option to set the preferred format explicitly, as relying on system locale may produce inconsistent results, and besides, users may have specific preferences to match their youtube theme or whatever.

p.s. otherwise, this extension is a real gem. Yet another piece of the "de-enshittification of youtube" puzzle. Thanks for it!

**Comment by RandomGamingDev** [comment]
> Is there a possibility to let the user specify an strftime()-compatible format? That would solve the issue completely for everyone.

That does sound like a good idea. I'll probably implement it when I do have time, although I might want to import another library for it since Javascript doesn't natively support it since it's mainly a Python function and all. Goes against a bit of the lightweight mentality, but it's probably fine.

> p.s. otherwise, this extension is a real gem. Yet another piece of the "de-enshittification of youtube" puzzle. Thanks for it!

Thanks! :D

**Comment by shapirus** [comment]
Staying within the native JS capabilities would be reasonable. It looks like JS is being JS in this case too: reinventing the wheel and being a special little flower instead of simply supporting the age-old standard C function (strftime).

I'm not a JS expert, but it seems that it does have its own date formatting function: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date

It has its own format: https://tc39.es/ecma262/multipage/numbers-and-dates.html#sec-date-time-string-format

This would work just as well as what I initially proposed, the actual format being different is no big deal at all.
